### ISSUE TITLE
s2wasm accept stdin?

### ISSUE THREAD

**Issue by zachrip** [WebAssembly/binaryen#1314]
Hello all, I'm trying to pipe into s2wasm but it doesn't seem to support that? I've gotten stdout working fine but at the moment I have to create a temp file from llc then call s2wasm with that. Am I doing something wrong? Does s2wasm support reading from stdin?

**Comment by kripken** [comment]
Yes, I think none of the commandline tools here support piping in content from stdin. I'm not sure how to implement that in a cross-platform way myself, but hopefully someone can figure it out, would be good to fix this.

**Comment by zachrip** [comment]
@kripken I'll take a look into it, I don't have a whole lot of experience in this area but would love to contribute.

**Comment by kripken** [comment]
That would be great!

For the relevant code, look e.g. at `src/tools/wasm-opt.cpp`, where it does `add_positional("INFILE"`. That's where we prepare to receive the input file as a positional argument (i.e. an argument that has no prefix before it, it's just the filename itself). Somehow we'd need to find a nice way to say that if that isn't specified, then we read it from `stdin`. If we need support code for that, it could be in `src/support/file*` perhaps.

**Comment by zachrip** [comment]
@kripken it seems that other things like clang just read from stdin if you have a lone dash: `clang -`. If I'm not mistaken there's not much to do in terms of cross-platform work as you're just reading from cin, no?

**Comment by kripken** [comment]
I'm not sure. But don't most tools read stdin even without the `-`, they somehow detect that there is info in stdin and they read it? I honestly have no idea how these things work :)

**Comment by tlively** [comment]
s2wasm no longer exists, and all our tools should now support reading from stdin if given the `-`.
